src/compiler/checker.ts

```diff
@@ -383,20 +383,42 @@ namespace ts {
             // return undefined if we can't find a symbol.
         }
 
-        /** Returns true if node1 is defined before node 2**/
-        function isDefinedBefore(node1: Node, node2: Node): boolean {
-            let file1 = getSourceFileOfNode(node1);
-            let file2 = getSourceFileOfNode(node2);
+        const enum RelativeLocation {
+            Unknown,
+            SameFileLocatedBefore,
+            SameFileLocatedAfter,
+            DifferentFilesLocatedBefore,
+            DifferentFilesLocatedAfter,
+        }
+
+        function isLocatedBefore(origin: Node, target: Node): boolean {
+            switch (getRelativeLocation(origin, target)) {
+                // unknown is returned with nodes are in different files and order cannot be determined based on compilation settings

```

Comment:
I assume this would happen if the files were external modules. Is there any other way this could happen (if the files are part of the global script)?